### DISCUSSION TITLE
Json array of object to HL7 segments

### DISCUSSION THREAD

**Discussion by AbboudyGeagea** [nextgenhealthcare/connect#5700] [Q&A] [answered]
Hello, 
I am trying to build an ORM message from Json file by converting it to HL7 using Mirth Connect V 4.1.0

till now I was able to build everything but I got stuck when it came to building multiple AL1 and DG1 segments from Json array of objects. 
What is happening (the last failure I am facing) is that I am able to read and write all the objects but the first one (let's say AL1) is in it's place (under PV1) but the others are being written at the end of the message (under OBX). 
Same applies to DG1. 

I will attach my channel and a test message.
I really appreciate anyone help in advance. 
[ORM_V2.txt](https://github.com/nextgenhealthcare/connect/files/10841823/ORM_V2.txt)
[GE_CVIS_ORM.zip](https://github.com/nextgenhealthcare/connect/files/10841827/GE_CVIS_ORM.zip)
![95c2df89-45a5-4a98-94af-e068e2a2c249](https://user-images.githubusercontent.com/32864471/221622398-d718b98c-cf25-4447-96d7-c229999103f8.jpg)

**Comment by pacmano1** [comment, marked as answer]
When you create a segment with createSegment it adds it to the end of the already existing message (tmp in your case).

Since you have an outbound message template the already existing message is the one you have in the template, so your createSegment's are adding after the OBX[1].   Your other assigments effectively "insert" into the existing outbound template, the ones with createSegment add at the end.
